src/Microsoft.Extensions.Options/IOptionsInitializer.cs

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.Extensions.Options
+{
+    public interface IOptionsInitializer<TOptions> where TOptions : class
+    {
+        /// <summary>
+        /// Initializes the options instance.
+        /// </summary>
+        /// <param name="instance">The options instance.</param>
+        /// <returns></returns>
+        TOptions Initialize(TOptions instance);
```

Comment:
Nit: The `TOptions` parameter in `IOptionsConfiguration.Configure()` is called `options`. I prefer `instance` and I think we should make it consistent.